forte/data/data_store.py

```diff
@@ -524,31 +558,48 @@ def get(
             for entry in entries:
                 yield entry
 
-    def next_entry(self, tid: int) -> List:
+    def next_entry(self, tid: int) -> Optional[List]:
         r"""Get the next entry of the same type as the ``tid`` entry.
         Call ``get_entry()`` to find the current index and use it to find
-        the next entry.
+        the next entry. If it is a non-annotation type, it will be sorted in
+        the insertion order, which means ``next_entry`` would return the next
+        inserted entry.
 
         Args:
             tid (int): Unique id of the entry.
 
         Returns:
-            The next entry of the same type as the ``tid`` entry.
+            A list of attributes representing the next entry of the same type
+            as the ``tid`` entry. Return `None` when accessing the next entry
+            of the last element in entry list.
         """
-        raise NotImplementedError
-
-    def prev_entry(self, tid: int) -> List:
+        _, entry_type = self.get_entry(tid=tid)
+        index_id: int = self.get_entry_index(tid=tid)
+        entry_list = self.__elements[entry_type]
+        if index_id >= len(entry_list) - 1:
+            return None
```

Comment:
I think the idea is to only return `None` at the edges. For indices that's out of the list, should still be `IndexError`